enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,205 @@
+
+### Phase 2: Move to Layered Update Path by Default 
+
+After On Clustering GA, we will gradually deprecate the current configuration method of direct file writing to disk and move to layered update path (Phase 1) by default. At the end of this phase, OCP nodes will be configured in a full image-based way and will have a RHEL-bootc layer, a CoreOS layer, an OCP node layer, and day-2 configuration layers added on top built and rolled out by On Cluster Layering discussed in Phase 1. Please note here that the proposal and implementation details of the first three layers (the RHEL-bootc layer, the CoreOS layer and the OCP node layer) are explained in the Splitting the RHCOS Image into layers  enhancement, thus a non-goal for this enhancement. Some of the gaps we observed here are:
+  * "Default" MCO managed configuration moved to /usr and shipped as a container image layer: Includes all MCO specific config rendered to /usr + MachineConfigs which are rendered to /etc -> /usr/etc (to be confirmed) 
+  * Configs that involve file writing to immutable local file system directories need to be handled: The immutable CoreOS model guards the immutability of certain directories. With that being said, if we write a file (e.g. SSH key) to an immutable directory during an OS image build and use rpm-ostree to rebase onto the newly built OS image, that file will not actually be there when the node reboots. 
+  * Address functionality gap between bootc and rpm-ostree: figure out a story for kernel argument changes as a container image layer/label 
+  * Carve a path for changes made to nodes without applying MachineConfig like changes made by other OCP operators (e.g. SRIO-V) or MCO cert writing.
+
+### Phase 3: Deprecate rpm-ostree and Move Layered Image Update Path to Bootc 
+
+Upon completion of the above, the last step is to update the image rollout mechanism. In the foreseeable future, we will be expecting a dual-stream RHCOS where rpm-ostree will be gradually deprecated and transition its responsibility to bootc. Following this trend, MCO will also enable image deployment by both “rpm-ostree rebase” and “bootc switch” and, in the end, settle down on “bootc switch” for image rollout. 
```

Comment:
This is referring to the point at which we are supporting both `rpm-ostree rebase` and `bootc switch` as we haven't completely deprecated rpm-ostree yet. IIRC from the discussions, we never plan to ship this interim phase to customers but this will be a phase we need to cross when implementing and testing out the move to bootc. That is why the final phase of all this is to do the final move to `bootc switch` after OCL is GA and layered update path is made the default.

Reply:
Proposal: instead of using the term "dual-stream" for this, which already has a meaning in OCP, let's use e.g. "dual support" consistently as used below?

Reply:
fixed